### PR TITLE
Reimplement 'EigenTEERegister' and 'OperatorWork'

### PR DESCRIPTION
**Pull request by Weber-H2O2**
TODO: Remove all `unwrap()` with a better error handling to make `fns` robust.

### CODE REVIEW

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/register_func.rs [hunk 1]
```diff
@@ -23,16 +23,32 @@ use std::str;
...
+
+use eigen_crypto::sign::ecdsa::{KeyPair, EcdsaKeyPair};
+use rand::Rng;
+use lazy_static::lazy_static;
+
+// TODO: For a naive implementation, now the key pair is never changed
+//       when `fns` is launched, however, in the future, the key pair
+//       may be updated by some mechanism
+lazy_static! {
+    static ref CACHED_KEY_PAIR: EcdsaKeyPair = {
```
Comment (eigmax) [inline cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/register_func.rs:35]: cache another AES key.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    fn execute(&mut self, _context: WorkerContext) -> Result<String> {
+        let input = self
+            .input
+            .take()
+            .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        let key_pair = register_func::get_key_pair();
+        let s1 = vec![];
+        let s2 = vec![];
+        let op1 = eigen_crypto::ec::suite_b::ecies::decrypt(key_pair, &input.cipher_op1.as_bytes(), &s1, &s2).unwrap();
```
Comment (eigmax) [inline cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs:128]: Decrypt AES Key by ECIES, the decrypt cipher oprand by AES key.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        };
+
+        let s1 = vec![];
+        let s2 = vec![];
+        let key_pair = register_func::get_key_pair();
+
+        let alg = &eigen_crypto::sign::ecdsa::ECDSA_P256_SHA256_ASN1;
+        let public_key = eigen_crypto::sign::ecdsa::UnparsedPublicKey::new(alg, key_pair.public_key());
+
+        let cipher = eigen_crypto::ec::suite_b::ecies::encrypt(&public_key, &s1, &s2, &result.to_be_bytes());
```
Comment (eigmax) [inline cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs:152]: encrypt result by AES Key.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs (hunk 2 above)
Comment (Weber-H2O2) [reply cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs]: OK. I've added these in the latest commitment.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/register_func.rs (hunk 1 above)
Comment (Weber-H2O2) [reply cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/register_func.rs:35]: OK. I've added these in the latest commitment.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs (hunk 3 above)
Comment (Weber-H2O2) [reply cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs:160]: OK. I've added these in the latest commitment.

**Comment by Weber-H2O2** [comment]
@eigmax Please review the latest commit: [c09b022](https://github.com/ieigen/ieigen/pull/59/commits/c09b022fbf286f82962accfe48e95f3f4f49e68f), I've add AES encrypt and decrypt at the corresponding places.

**Comment by Weber-H2O2** [comment]
TODO: all algorithm operators now return a base64 encoding string, however, `ecall` now only supports `uint256`. We should use a property method to work around.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+            worker_id: 0,
+            func_name: "operator".to_string(),
+            input: None,
+        }
+    }
+}
+
+enum OperatorKind {
+    Add,
+    Add1,
```
Comment (eigmax) [inline cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs:50]: xx1 is not easy to understand. How about names like add_cipher_cipher,  add_cipher_plain?

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    Sub1,
+    Enc,
+    Des,
+}
+
+struct OperatorWorkerInput {
+    // TODO: Use a more property field name to describe what we want to save here.
+    //       e.g., when operator is `Enc`, `cipher_op1` is not a cipher one
+    op: OperatorKind,
+    cipher_op1: String,
```
Comment (eigmax) [inline cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs:61]: op usually represents for operators, the operand is easier to understand.

File: cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+        // now `op` may be 'add' or 'sub'
+        // number is parsed with big endian
+
+        let splited = Vec::from_iter(msg.split(",").map(String::from));
+
+        let op = splited.get(0).unwrap();
+
+        match op.as_str() {
+            "add" => {
+                let op_num = splited.get(1).unwrap().parse::<u64>().unwrap();
```
Comment (eigmax) [inline cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs:91]: `let vec: Vec<&str> = msg.split(",").collect(); `  then you can use index to access the value after the arguments number check.

**Pull request closed without merging**